src/atoms/affine/partialtranspose.jl

```diff
@@ -0,0 +1,104 @@
+import Base.sign
+export partialtranspose, PartialTransposeAtom
+export sign, curvature, monotonicity, evaluate, conic_form!
+
+struct PartialTransposeAtom <: AbstractExpr
+    head::Symbol
+    id_hash::UInt64
+    children::Tuple{AbstractExpr}
+    size::Tuple{Int, Int}
+    sys::Int
+    dims::Vector
+
+    function PartialTransposeAtom(x::AbstractExpr, sys::Int, dims::Vector)
+        if x.size[1] ≠ x.size[2]
+            throw(ArgumentError("Only square matrices are supported"))
+        end
+        if ! (1 ≤ sys ≤ length(dims))
+            throw(ArgumentError("Invalid system, should between 1 and ", length(dims), "; got ", sys))
+        end
+        if x.size[1] ≠ prod(dims)
+            throw(ArgumentError("Dimension of system doesn't correspond to dimension of subsystems"))
+        end
+        children = (x, )
+        return new(:partialtranspose, hash(children), children, x.size, sys, dims)    
+    end
+end
+
+function sign(x::PartialTransposeAtom)
+    return sign(x.children[1])
+end
+
+function curvature(x::PartialTransposeAtom)
+    return ConstVexity()
+end
+
+function monotonicity(x::PartialTransposeAtom)
+    return (Nondecreasing(),)
+end
+
+function evaluate(x::PartialTransposeAtom)
+    return partialtranspose( evaluate( x.children[1] ))
+end
+
+"""
+    partialtranspose(x, sys::Int, dims::Vector)
+
+Returns the partial transpose of `x` over the `sys`th system, where `dims` is a vector of integers encoding the dimensions of each subsystem.
+"""
+function partialtranspose(x::AbstractMatrix, sys::Int, dims::Vector) 
+    if size(x,1) ≠ size(x,2)
+            throw(ArgumentError("Only square matrices are supported"))
+    end
+    if ! (1 ≤ sys ≤ length(dims))
+            throw(ArgumentError("Invalid system, should between 1 and $(length(dims)); got $sys"))
+    end
+    if size(x,1) ≠ prod(dims)
+            throw(ArgumentError("Dimension of system doesn't correspond to dimension of subsystems"))
+    end
+    n = length(dims)
+    d = prod(dims)
+    s = n - sys + 1
+    p = collect(1:2n)
+    p[s] = n + s
+    p[n + s] = s
+
+    rdims = reverse(dims)
+    r = reshape(x, [rdims;rdims]...)
+    return reshape(permutedims(r,p),(d,d))
+end
+
+
+function permutedims_matrix(dims, p)
```

Comment:
Let's add the docstring from https://github.com/ericphanson/QuantumSDPs.jl/blob/97225a7616f08f46de568091dcf63956b2125b1c/src/choi2so.jl#L4-L11

Reply:
done.